internal/command/flags/urfave.go

```diff
@@ -0,0 +1,60 @@
+package flags
+
+import (
+	"github.com/urfave/cli/v2"
+
+	"github.com/weaveworks/flintlock/internal/config"
+	"github.com/weaveworks/flintlock/pkg/defaults"
+)
+
+type WithFlagsFunc func() []cli.Flag
+
+func CLIFlags(options ...WithFlagsFunc) []cli.Flag {
+	flags := []cli.Flag{}
+
+	for _, group := range options {
+		flags = append(flags, group()...)
+	}
+
+	return flags
+}
+
+func WithContainerDFlags() WithFlagsFunc {
+	return func() []cli.Flag {
+		return []cli.Flag{
+			&cli.StringFlag{
+				Name:  containerdSocketFlag,
+				Value: defaults.ContainerdSocket,
+				Usage: "The path to the containerd socket.",
+			},
+			&cli.StringFlag{
+				Name:  containerdNamespace,
+				Value: defaults.ContainerdNamespace,
+				Usage: "The name of the containerd namespace to use.",
+			},
+		}
+	}
+}
+
+func WithHTTPEndpointFlags() WithFlagsFunc {
+	return func() []cli.Flag {
+		return []cli.Flag{
+			&cli.StringFlag{
+				Name:  httpEndpointFlag,
+				Value: defaults.HTTPAPIEndpoint,
+				Usage: "The endpoint for the HTTP server to listen on.",
+			},
+		}
+	}
+}
+
+func ParseFlags(cfg *config.Config) cli.BeforeFunc {
```

Comment:
It will parse everything we know, and if not defined it will be empty or default value.